lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -339,5 +344,15 @@ def dlq_enabled?
       respond_to?(:execution_context) && execution_context.respond_to?(:dlq_writer) &&
         !execution_context.dlq_writer.inner_writer.is_a?(::LogStash::Util::DummyDeadLetterQueueWriter)
     end
+
+    EMPTY_HASH = {}.freeze
+
+    def dig_value(val, *keys)
```

Comment:
this implementation won't work if the thing at the key being dug is actually an empty hash -- it'll return `nil` instead of the object that exists at that address.

Although our one specific use is not likely to hit this issue (or if it does, we already have other problems), I'm wary of implementing a `dig` that behaves differently than the one we know and love from Ruby 2.3+.

Here is an implementation that borrows heavily from [ruby backports](https://github.com/marcandre/backports/blob/71bd2a5ba0c64e1f7ef90503d3f7d3ea167b5028/lib/backports/2.3.0/hash/dig.rb) without monkey-patching `Hash`:

~~~ ruby
def dig_value(val, first_key, *rest_keys)
  fail(TypeError, "cannot dig value from #{val.class}") unless val.kind_of?(Hash)
  val = self[first_key]
  return val if rest_keys.empty? || val == nil
  dig_value(val, *rest_keys)
end
~~~